corehq/apps/importer/util.py

```diff
@@ -423,28 +428,39 @@ def is_valid_id(uploaded_id, domain, cache):
     )
 
 
-def get_id_from_name(uploaded_name, domain, cache):
+def get_id_from_name(name, domain, cache):
     '''
-    :param uploaded_name: A username or group name
+    :param name: A username, group name, or location name/site_code
     :param domain:
     :param cache:
     :return: Looks for the given name and returns the corresponding id if the
     user or group exists and None otherwise. Searches for user first, then
-    group.
+    group, then location
     '''
-    if uploaded_name in cache:
-        return cache[uploaded_name]
-    try:
-        name_as_address = uploaded_name
-        if '@' not in name_as_address:
-            name_as_address = format_username(uploaded_name, domain)
-        user = CouchUser.get_by_username(name_as_address)
-        id = getattr(user, 'couch_id', None)
-    except NoResultFound:
-        id = None
-    if not id:
-        group = Group.by_name(domain, uploaded_name, one=True)
-        id = getattr(group, 'get_id', None)
-
-    cache[uploaded_name] = id
+    if name in cache:
+        return cache[name]
+
+    def get_from_user(name):
+        try:
+            name_as_address = name
+            if '@' not in name_as_address:
+                name_as_address = format_username(name, domain)
+            user = CouchUser.get_by_username(name_as_address)
+            return getattr(user, 'couch_id', None)
+        except NoResultFound:
+            return None
+
+    def get_from_group(name):
+        group = Group.by_name(domain, name, one=True)
+        return getattr(group, 'get_id', None)
+
+    def get_from_location(name):
+        for filter_ in [{'site_code': name}, {'name__iexact': name}]:
+            try:
+                return SQLLocation.objects.get(domain=domain, **filter_).location_id
+            except SQLLocation.DoesNotExist:
+                pass
+
+    id = get_from_user(name) or get_from_group(name) or get_from_location(name)
```

Comment:
i wonder if we want to consider namespacing this somehow (like `"location:Boston"`), or at least documenting the order somewhere externally. seems like we will eventually need a way for people to explicitly specify which one they want to associate in case of duplicate names (using different columns in the sheet could also work).

@sheelio have you thought about this much? right now it will just use the first user then group then location it finds with a name match which 95% of the time is probably what you want, just confusing.